daemon/oci_windows.go

```diff
@@ -25,11 +25,31 @@ func (daemon *Daemon) createSpec(c *container.Container) (*specs.Spec, error) {
 	// In base spec
 	s.Hostname = c.FullHostname()
 
+	if err := daemon.setupSecretDir(c); err != nil {
+		return nil, err
+	}
+
 	// In s.Mounts
 	mounts, err := daemon.setupMounts(c)
 	if err != nil {
 		return nil, err
 	}
+
+	if !c.HasBeenStartedBefore && len(c.SecretReferences) > 0 {
```

Comment:
I am not sure this is the best place to explain when to mount containers on Windows, in general, but perhaps you can give me a rough idea of the comment you would like to see. Thanks for pointing me to `HasBeenStartedBefore`; this check fixes an issue where the code was trying to recreate the symlinks for a container that had already been started, which fails.